src/code/z_skelanime.c

```diff
@@ -1451,7 +1451,7 @@ BAD_RETURN(s32) SkelAnime_Init(PlayState* play, SkelAnime* skelAnime, SkeletonHe
         skelAnime->morphTable =
             ZELDA_ARENA_MALLOC(skelAnime->limbCount * sizeof(*skelAnime->morphTable), "../z_skelanime.c", 2969);
     } else {
-        ASSERT(limbCount == skelAnime->limbCount, "joint_buff_num == this->joint_num", "../z_skelanime.c", 2973);
+        ASSERT(tableLength == skelAnime->limbCount, "joint_buff_num == this->joint_num", "../z_skelanime.c", 2973);
```

Comment:
I'm going to have to disagree with using ARRAY_LENGTH to pass in the limb count, because of this line right here. This assert shows that tableLength should be coupled to limbCount, and not necessarily the array length.

In MM, you have z_en_invadepoh.c which manages multiple different NPC characters with incompatible animated models, all under the `ACTOR_EN_INVADEPOH` instance. In this case, jointTable and morphTable are sized to the largest character, but will call `SkelAnime_InitFlex` with the different `*_LIMB_MAX` values.

Reply:
their original name is `joint_buff_num`, aka the length of the joint buffers (the table in question).
However your point about usage in MM is a fair one to consider.

Reply:
It's weird, because even if that is how it was implemented, the code still doesn't fully verify correctness because it assumes morphTable is the same length as jointTable. For that, it seems better to keep using `*_LIMB_MAX` because it's at least easier to understand.

Reply:
Yeah it seems like generally speaking the joint buffer is the "important" one. Its the only one that gets checked for dynamic allocation, for example.

Ultimately I see both approaches here as valid, I didnt necessarily have a problem with limb_max being used as the value.

Regardless of what gets chosen here I think it may be a good idea to name our "limb" enums more appropriately after the buffer indicies. That is, the fact that it has the root translation information which bumps every index up by one.

Reply:
> This assert shows that tableLength should be coupled to limbCount, and not necessarily the array length.

Well yes the array length should be coupled to the limb count, it has to store a limb-count-amount of elements

To me a further hint that this check is meant to check the length of the provided arrays is that 0 is passed for dynamically allocated arrays and the value is not checked then

Reply:
> To me a further hint that this check is meant to check the length of the provided arrays is that 0 is passed for dynamically allocated arrays and the value is not checked then

I don't doubt this; just question whether it's reasonable to use that pattern when the assert is "bugged"

Reply:
It does work in 98% of calls, for reference the only spots that pass a length == limbCount < `ARRAY_COUNT(jointTable)` are
- https://github.com/Dragorn421/oot/blob/d50962f22d5228aecfd6d6f3946159b98dd94253/src/overlays/actors/ovl_Boss_Sst/z_boss_sst.c#L345-L351
- https://github.com/Dragorn421/oot/blob/d50962f22d5228aecfd6d6f3946159b98dd94253/src/overlays/actors/ovl_En_Dnt_Nomal/z_en_dnt_nomal.c#L202-L203
- https://github.com/Dragorn421/oot/blob/d50962f22d5228aecfd6d6f3946159b98dd94253/src/overlays/actors/ovl_En_Poh/z_en_poh.c#L935-L936